index.js

```diff
@@ -52,7 +52,7 @@ function ensureAllCommentsPrinted(astComments) {
 function format(text, opts, addAlignmentSize) {
   addAlignmentSize = addAlignmentSize || 0;
 
-  const ast = parser.parse(text, opts);
+  const ast = opts.astTransform(parser.parse(text, opts));
```

Comment:
Yea, that totally makes sense. The parser could be made pluggable in other ways. I would however still pass in the options.